pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -339,6 +339,53 @@ func TestEtcdUpdate(t *testing.T) {
 
 }
 
+func TestNoOpUpdates(t *testing.T) {
+	server, registry := NewTestGenericEtcdRegistry(t)
```

Comment:
All the other tests are testing updates against new records made with the current version which will not set resourceVersion in etcd. I want to make sure when this fix runs against an etcd containing old incorrect records which contain resourceVersion, we get correct update behavior